qupath-gui-fx/src/main/java/qupath/lib/gui/charts/Charts.java

```diff
@@ -834,6 +823,268 @@ public static ScatterchartBuilder scatterChart() {
+	 * @return the builder
+	 */
+	public static BarChartBuilder barChart() {
+		return new BarChartBuilder();
+	}
+
+	abstract static class XYCategoryChartBuilder<T extends XYCategoryChartBuilder<T, S>, S extends XYChart<String, Number>> extends XYChartBuilder<T, S, String, Number> {
+
+		protected abstract S createNewChart(Axis<String> xAxis, Axis<Number> yAxis);
+
+		private Double yLower, yUpper;
+
+		/**
+		 * Set the lower bound for the y-axis.
+		 * @param lowerBound
+		 * @return this builder
+		 */
+		public T yAxisMin(double lowerBound) {
+			this.yLower = lowerBound;
+			return getThis();
+		}
+
+
+		/**
+		 * Set the upper bound for the y-axis.
+		 * @param upperBound
+		 * @return this builder
+		 */
+		public T yAxisMax(double upperBound) {
+//			this.xUpper = upperBound;
+			return getThis();
+		}
+
+		/**
+		 * Set the lower and upper bounds for the y-axis.
+		 * @param lowerBound
+		 * @param upperBound
+		 * @return this builder
+		 */
+		public T yAxisRange(double lowerBound, double upperBound) {
+			this.yLower = lowerBound;
+			this.yUpper = upperBound;
+			return getThis();
+		}
+
+
+		@Override
+		protected S createNewChart() {
+			var xAxis = new CategoryAxis();
+			var yAxis = new NumberAxis();
+
+			setBoundIfValid(yAxis.lowerBoundProperty(), yLower);
+			setBoundIfValid(yAxis.upperBoundProperty(), yUpper);
+
+			if (xLabel != null)
+				xAxis.setLabel(xLabel);
+			if (yLabel != null)
+				yAxis.setLabel(yLabel);
+			return createNewChart(xAxis, yAxis);
+		}
+
+		private static void setBoundIfValid(DoubleProperty prop, Double val) {
+			if (val != null && Double.isFinite(val))
+				prop.set(val);
+		}
+
+	}
+
+	/**
+	 * Builder for creating scatter charts.
+	 */
+	public static class BarChartBuilder extends XYCategoryChartBuilder<BarChartBuilder, BarChart<String, Number>> {
+
+		private final ObservableList<Series<String, Number>> series = FXCollections.observableArrayList();
+		private final ObservableList<PathObject> pathObjects = FXCollections.observableArrayList();
+
+		private BarChartBuilder() {}
+
+		@Override
+		protected String getDefaultWindowTitle() {
+			return "Bar Chart";
+		}
+
+		/**
+		 * Plot values extracted from objects within a specified collection.
+		 * @param name
+		 * @param <T>
+		 * @name the name of the data series (useful if multiple series will be plotted, otherwise may be null)
+		 * @param collection the objects to plot
+		 * @param xFun function capable of extracting a numeric value for the x location from each object in the collection
+		 * @return this builder
+		 */
+		public <T> BarChartBuilder series(String name, Collection<? extends T> collection, Function<T, PathClass> xFun) {
+			var classAndCount = collection
+					.stream().map(xFun)
+					.collect(
+							Collectors.groupingBy(Function.identity(), Collectors.counting())
+					);
+			return series(name, classAndCount.entrySet().stream()
+					.map(e -> new Data<>(e.getKey().getName(), (Number)e.getValue(), e.getKey()))
```

Comment:
Using `PathClass.getName()` just gets one component of the classification - you need `PathClass.toString()` to get all of them.
https://qupath.readthedocs.io/en/stable/docs/concepts/classifications.html
But this will also fail with unclassified objects (unless you replace `null` with [`PathClass.NULL_CLASS`](https://github.com/qupath/qupath/blob/ae4cc88429aea647b09bbd380782bc206237800b/qupath-core/src/main/java/qupath/lib/objects/classes/PathClass.java#L156)).

Reply:
I think I've handled this, but I'm not exactly confident I've done it the right way